### PR TITLE
aria2: add config file & plist

### PR DESCRIPTION
**Pull request by icpz**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

So that we can use ```brew services``` manage aria2 now.

### CODE REVIEW

File: Formula/aria2.rb [hunk 1]
```diff
@@ -37,9 +37,42 @@ def install
     system "./configure", *args
     system "make", "install"
 
+    (buildpath/"aria2.conf").write <<~EOS
+      enable-rpc=true
+      rpc-secret=barfoo
```
Comment (fxcoudert) [inline Formula/aria2.rb]: Hardcoding a secret in all installs sounds bad.

**Comment by fxcoudert** [comment]
If upstream doesn't ship a configuration, I do not think we should create our own.

**Comment by lcdtyph** [comment]
But if we don't provide a config file, how to write the plist?

File: Formula/aria2.rb (hunk 1 above)
Comment (lcdtyph) [reply Formula/aria2.rb]: How about add a ```caveats``` block to inform users to create a configuration by themselves?

**Comment by fxcoudert** [comment]
If it doesn't run out of the box, then it's probably meant for power users?

**Comment by lcdtyph** [comment]
Actually, the configuration is mostly need by geeks. A normal user could use it without config just like wget.

**Comment by lcdtyph** [comment]
How about now

**Comment by lcdtyph** [comment]
@fxcoudert 
Any advice please?

File: Formula/aria2.rb [hunk 2]
```diff
@@ -40,6 +40,35 @@ def install
     bash_completion.install "doc/bash_completion/aria2c"
   end
 
+  def caveats; <<~EOS
+    aria2 requires a config file to start as daemon.
+    Please create #{etc}/aria2.conf and edit your custom configuration.
```
Comment (SMillerDev) [inline Formula/aria2.rb]:
```suggestion
    Please create #{etc}/aria2.conf and add a configuration.
```

**Comment by lcdtyph** [comment]
Any advice?

**Comment by zbeekman** [comment]
I am personally 👎 for providing an `aria2` daemon plist. (I'm concerned about unruly daemons that users forget about that end up having bugs, provide a larger attack surface etc.)

**Comment by lcdtyph** [comment]
I don't think that is a problem... Same reason could also be applied to nginx, php, vsftpd etc., mis-configured which could provide an attack surface too.

**Pull request closed without merging**